taca/element/Element_Runs.py

```diff
@@ -0,0 +1,1191 @@
+        # Make a run subdir named after the zip file and extract manifests there
+
+        # Extract the contents of the zip file into the destination directory
+        unzipped_manifests = []
+        with zipfile.ZipFile(zip_src_path, "r") as zip_ref:
+            for member in zip_ref.namelist():
+                # Extract each file individually into the destination directory
+                filename = os.path.basename(member)
+                if filename:  # Skip directories
+                    source = zip_ref.open(member)
+                    target = open(os.path.join(self.run_dir, filename), "wb")
+                    unzipped_manifests.append(target.name)
+                    with source, target:
+                        target.write(source.read())
+
+        # Pick out the manifest to use
+        self.lims_manifest = [
+            m for m in unzipped_manifests if re.match(r".*_untrimmed\.csv$", m)
+        ][0]
+
+    def make_demux_manifests(
+        self, manifest_to_split: os.PathLike, outdir: os.PathLike | None = None
+    ) -> list[str]:
+        """Derive composite demultiplexing manifests
+        from a single information-rich manifest.
+        """
+
+        # Read specified manifest
+        with open(manifest_to_split) as f:
+            manifest_contents = f.read()
+
+        # Get '[SAMPLES]' section
+        split_contents = manifest_contents.split("[SAMPLES]")
+        assert (
+            len(split_contents) == 2
+        ), f"Could not split sample rows out of manifest {manifest_contents}"
+        sample_section = split_contents[1].strip().split("\n")
+
+        # Split into header and rows
+        header = sample_section[0]
+        sample_rows = sample_section[1:]
+
+        # Convert to list of dicts
+        sample_dicts = []
+        for row in sample_rows:
+            row_dict = dict(zip(header.split(","), row.split(",")))
+            sample_dicts.append(row_dict)
+
+        # Convert to dataframe
+        df = pd.DataFrame.from_dict(sample_dicts)
+
+        # Separate samples from controls
+        df_samples = df[df["Project"] != "Control"].copy()
+        df_controls = df[df["Project"] == "Control"].copy()
+
+        # Add bool indicating whether UMI is present
+        df_samples["has_umi"] = df_samples["Index2"].str.contains("N")
+
+        # Add masks
+        df_samples["I1Mask"] = df_samples["Index1"].apply(
+            lambda seq: get_mask(
+                seq=seq,
+                keep_Ns=False,
+                prefix="I1:",
+                cycles_used=self.cycles["I1"],
+            )
+        )
+        df_samples["I2Mask"] = df_samples["Index2"].apply(
+            lambda seq: get_mask(
+                seq=seq,
+                keep_Ns=False,
+                prefix="I2:",
+                cycles_used=self.cycles["I2"],
+            )
+        )
+        df_samples["UmiMask"] = df_samples["Index2"].apply(
```

Comment:
Please also include the case that UMIs are included in `I1`. For example, the IDT UMI indexes has a format "XXXXXXXXXXNNNNNNNNN-XXXXXXXXXX"

Reply:
I thought you said UMIs were always in `I2`, but we can implement it for both :)

Reply:
Sorry but it was a missunderstanding then! The UMIs for scATAC-seq are always the entire I2. But for other RNA-seq and DNA-seq libraries the UMIs are part of I1 or I2.

Reply:
How's this?
https://github.com/ssjunnebo/TACA/pull/16

Reply:
Looks great! Thanks!